test/models/orderbook.spec.ts

```diff
@@ -0,0 +1,31 @@
+import {Orderbook} from "../../src/orderbook";
+import {assert} from "chai";
+import "mocha";
+
+describe("Orderbook", () => {
+  it("cummulates bids and asks into sorted json", () => {
+    const orderbook = new Orderbook("USDC", "DAI");
+    orderbook.addAsk(1.1, 100);
+    orderbook.addAsk(1.2, 200);
+    orderbook.addAsk(1.01, 300);
+
+    orderbook.addBid(0.9, 50);
+    orderbook.addBid(0.99, 70);
+    orderbook.addBid(0.9, 30);
+
+    assert.equal(
+      orderbook.toJSON(),
+      JSON.stringify({
+        bids: [
+          [0.9, 80],
```

Comment:
Are you sure you want to use this structure for the bid. It seems compact, but an object like approach is more readable and extendable.

```js
asks: [
          { price: 1.01, volume: 300},
          { price: 1.1, volume: 100},
          ...
]
```

Reply:
One example for why would you want to use an object style representation is, if you use this for creating a endpoint or a graph, you might want to also show how many orders are in a price point, or any other relevant information.

Reply:
The reason I used this representation is that it is the one used in the visualization library (https://6jxh4.csb.app/). It's also what the [poliniex api](https://poloniex.com/public?command=returnOrderBook&currencyPair=BTC_ETH&depth=50) returns. I agree that naming price/volume can be useful for comprehension and type safety. 

I don't think the number of orders at a given price point is relevant for an order book (the depth is).

Reply:
Not a big deal for me. If it's simpler for your targeted use case, I'm good

Reply:
@fleupold this is for taking a snapshot of the orderbook at our current moment in time, right?

We don't want to take validity into account. I mean, a more advanced version of this could keep track of the orders and return the book for a given batch. I understand that's not your current goal.

It can also be a service  that returns the book for a given batchId (future or past)

Reply:
It depends what is your input here. If you are e.g. using the BatchExchangeViewer.sol contract you can chose for which "target batch" you want your orders and availableSellBalances. This way you can make it behave like you suggested. 
The "orderbook logic" is independent of its data source (could even come from the graph)